### PR TITLE
Improve support for electric tank water heaters

### LINKED ISSUE #68: Add Support for Electric Water Heaters
We should add support for Electric (non Heat Pump) Water Heaters.

### PR DESCRIPTION
**Pull request by tronikos**
Fixes #68

1. Removes non working HOTWATER,
2. Removes non working POWRWATT,
3. Adds read only WHTRMODE,
4. Adds support for presets: Energy Saver (0), Performance (1) via WHTRCNFG

### CODE REVIEW

File: econet_electric_tank_water_heater.yaml [hunk 1]
```diff
@@ -63,3 +56,7 @@ text_sensor:
     name: "Heating Element State"
     sensor_datapoint: HEATCTRL
     icon: "mdi:heating-coil"
+  - platform: econet
+    name: "Mode"
+    sensor_datapoint: WHTRMODE
+    entity_category: "diagnostic"
```
Comment (barndawgie) [inline econet_electric_tank_water_heater.yaml:62]: Can this not be set via the Climate entity like with the other devices? If so, this sensor seems redundant. If not, I wonder if it should not be a diagnostic.

Comment (tronikos) [reply econet_electric_tank_water_heater.yaml:62]: From the discord discussion WHTRMODE was: Off-No Demand (0), Off-Delay Tmr(1), Vacation (11) and it seems to be read only. They were unsuccessful in modifying it, contrary to the WHTRCNFG that worked for them.

Comment (barndawgie) [reply econet_electric_tank_water_heater.yaml:62]: Can we not get the vacation mode added in? Weird that it's non-contiguous...

Comment (tronikos) [reply econet_electric_tank_water_heater.yaml:62]: Added in the presets? Mark was unsuccessful setting it via either WHTRMODE or WHTRCNFG.

Comment (tronikos) [reply econet_electric_tank_water_heater.yaml:62]: This text sensor will just show the string value as returned. My guess is there are other modes between 1 and 11, like heating upper, heating lower, etc.

Comment (barndawgie) [reply econet_electric_tank_water_heater.yaml:51]: Alright, well, this seems like an improvement over what's checked in currently.

**Review by barndawgie — APPROVED** [review]
LGTM

**Pull request merged**